From 539f50afd56a0eea57aa0c965d9406731a0c0f1d Mon Sep 17 00:00:00 2001
From: Marian Chereji <marian.chereji@freescale.com>
Date: Wed, 4 Sep 2013 11:39:23 +0300
Subject: [PATCH 114/130] dpa_offload: Fix static index initialization in DPA
 Classifier

The static indexes used in recursive functions "init_hm_chain" and
"remove_hm_chain" were not initialized. Also, "remove_hm_chain" was
not decrementing this index as it removed header manipulations.

Signed-off-by: Marian Chereji <marian.chereji@freescale.com>
Change-Id: I00bd44c00749578a4c785741082e0ad2c1076c27
Reviewed-on: http://git.am.freescale.net:8181/4412
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Bulie Radu-Andrei-B37577 <Radu.Bulie@freescale.com>
Reviewed-by: Rivera Jose-B46482 <German.Rivera@freescale.com>
[Fetch from http://git.freescale.com/git/cgit.cgi/ppc/sdk/linux.git/,
 Tag: fsl-sdk-v1.4.5
 rebase on current context.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/dpa_classifier.c |    6 ++++--
 1 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/dpa_classifier.c b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
index f8ed8b7..751dacd 100644
--- a/drivers/staging/fsl_dpa_offload/dpa_classifier.c
+++ b/drivers/staging/fsl_dpa_offload/dpa_classifier.c
@@ -3585,7 +3585,7 @@ static int init_hm_chain(void *fm_pcd, struct list_head *chain_head,
 	t_Error error;
 	struct dpa_cls_hm_node *pcurrent, *pnext;
 	t_FmPcdManipParams params;
-	static int index;
+	static int index = 0;
 	static int num_int_nodes;
 
 	BUG_ON(!chain_head);
@@ -3798,7 +3798,7 @@ int remove_hm_chain(struct list_head *chain_head, struct list_head *item)
 	int err = 0;
 	struct dpa_cls_hm_node *pcurrent;
 	t_Error error;
-	static int index;
+	static int index = 0;
 
 	BUG_ON(!chain_head);
 	BUG_ON(!item);
@@ -3829,6 +3829,8 @@ int remove_hm_chain(struct list_head *chain_head, struct list_head *item)
 
 	remove_hm_node(pcurrent);
 
+	index--;
+
 	return err;
 }
 
-- 
1.7.5.4

